modules/xmpp/JingleSessionPC.js

```diff
@@ -799,32 +796,30 @@ export default class JingleSessionPC extends JingleSession {
             for (const localTrack of localTracks) {
                 this.peerconnection.addTrack(localTrack);
             }
-            this.peerconnection.createOffer(
-                offerSdp => {
-                    this.peerconnection.setLocalDescription(
-                        offerSdp,
-                        () => {
+            this.peerconnection.createOffer(this.mediaConstraints)
+                .then(offerSdp => {
+                    this.peerconnection.setLocalDescription(offerSdp)
+                        .then(() => {
                             // NOTE that the offer is obtained from
                             // the localDescription getter as it needs to go
                             // though the transformation chain.
                             this.sendSessionInitiate(
                                 this.peerconnection.localDescription.sdp);
                             finishedCallback();
-                        },
-                        error => {
+                        })
+                        .catch(error => {
```

Comment:
hmm I'm not sure if using .catch is what we're looking for here (and other places). Wouldn't that also catch an eventual failure in then() as well ? It seems to me that more direct translation would be with then(success, failure) callbacks instead of then().catch() chains:
`this.peerconnection.setLocalDescription(offerSdp).then(() => { all good }, error => { ups failure });`

Reply:
You are right!